lib/rules/prefer-user-event.ts

```diff
@@ -90,59 +90,38 @@ export default ESLintUtils.RuleCreator(getDocsUrl)<Options, MessageIds>({
   },
   defaultOptions: [{ allowedMethods: [] }],
 
-  create(context, [options]) {
+  create(context, [options], helpers) {
     const { allowedMethods } = options;
     const sourceCode = context.getSourceCode();
-    let hasNamedImportedFireEvent = false;
-    let hasImportedFireEvent = false;
-    let fireEventAlias: string | undefined;
-    let wildcardImportName: string | undefined;
 
     return {
-      // checks if import has shape:
-      // import { fireEvent } from '@testing-library/dom';
-      ImportDeclaration(node: TSESTree.ImportDeclaration) {
-        if (!hasTestingLibraryImportModule(node)) {
-          return;
-        }
-        const fireEventImport = node.specifiers.find(
-          (node) =>
-            isImportSpecifier(node) && node.imported.name === 'fireEvent'
-        );
-        hasNamedImportedFireEvent = !!fireEventImport;
-        if (!hasNamedImportedFireEvent) {
+      ['CallExpression > MemberExpression'](node: TSESTree.MemberExpression) {
+        if (!helpers.getIsTestingLibraryImported()) {
```

Comment:
You don't have to do this anymore. Detection module is taking care of this and prevent the rules listeners you declare here from being executed if not testing library imported or filename not valid. You can find that at the very bottom of `detectTestingLibraryUtils`:

```javascript
if (helpers.canReportErrors() && ruleInstructions[instruction]) {
  return ruleInstructions[instruction](node);
}
```

Reply:
yeah I remembered about it when I went to sleep, but the laptop was already turned off 😆  but thanks for commenting it 👍

Reply:
removed!